frontend/src/pages/ErrorsV2/ErrorInstance/ErrorInstance.tsx

```diff
@@ -350,7 +350,6 @@ const ErrorInstance: React.FC<Props> = ({ errorGroup }) => {
 								if (logCursor) {
 									navigate(
```

Comment:
@Vadman97 - what happens if make this a proper link instead (see [comment](https://github.com/highlight/highlight/pull/4674#discussion_r1145623572))?

Reply:
I don't understand the issue then. Is the issue a problem with linking from the errors page or something specific about the logs page?

If it's the former, shouldn't we update the `navigate` statement right below this one as well?
If it's the latter, can you show a loom of what you mean about the `?query=` being the issue?

Reply:
here's a simple example of the problem:
* visit https://app.highlight.io/1/logs?query=foo
* the query `foo` is not in the search
hence, the problem doesn't have to do with this navigation itself. the issue is how we update the state from the search form with the query string, resetting the initial value even if one is set.
what's happening with the PR
1. the `query` value is parsed from the URL by `LogsPage`
2. the `query` value is passed to `SearchForm` as the`initialQuery`
3. on first render, `SearchForm` calls `setQuery('')` resetting the query and the query string to `query=''`

Reply:
switched this to use a `LinkButton` - shout out to @ccschmitz for making this so easy

Reply:
> on first render, SearchForm calls setQuery('') resetting the query and the query string to query=''

Sorry, where is this exactly?
I'm seeing this but I'm not sure if that's what you're talking about: https://github.com/highlight/highlight/blob/014061ae28ef8cda077e7771f2ad925f6c02219e/frontend/src/pages/LogsPage/SearchForm/SearchForm.tsx#L195-L201

Sorry if I'm digging in too much. Really want us to avoid the complexity difficulties we have with the session/error page because we have too much state and useEffects updating things.

Reply:
no worries! the only change is https://github.com/highlight/highlight/blob/796462c79015b95be4310b18e019dafcf1858ce4/frontend/src/pages/LogsPage/SearchForm/SearchForm.tsx#L71-L75

as i found with some debugging, `formState.useSubmit` triggers on initial render of the ariakit form.

before the change
```
	formState.useSubmit(() => {
		onFormSubmit(formState.values.query)
	})
```
would call `onFormSubmit('')` since the search form is not filled out, even when the `initialQuery` was provided

after the change
```
	formState.useSubmit(() => {
		if (formState.submitting) {
			onFormSubmit(formState.values.query)
		}
	})
```
checking for `formState.submitting` only calls the `onFormSubmit` when the ariakit form detects a 'submission' - aka a real input followed by `Enter` key or the input losing focus.

Reply:
Thanks for sharing the write up @Vadman97! Appreciate the writeup.

> as i found with some debugging, formState.useSubmit triggers on initial render of the ariakit form.

If I understand this correctly, we are saying `useSubmit` is being triggered when the component mounts? And just to be clear, it's that we are manually calling this -- I'm fairly certain ariakit's `useSubmit` doesn't do this by default (here's a [codesandbox](https://codesandbox.io/s/8sw6dv?module=src/form.tsx))

With that being said, I think there's something else strange going on here with these two `useEffect`s:
Here's a loom to describe what I mean:

https://www.loom.com/share/94fd8b66468f487ca8e6b7195a090c7e

Reply:
Actually...removing this `useEffect` seems to fix the issue:

https://github.com/highlight/highlight/blob/dae0318e3b4ea23bd06244364ca26324493798e2/frontend/src/pages/LogsPage/SearchForm/SearchForm.tsx#L195-L201

I'm not sure why we have this exactly. Sasha added it here but there's not a whole lot of context as to why:

https://github.com/highlight/highlight/pull/4524/files#diff-f847a8411b2f43cb4537fcb5260ba0125fbd6de14f055f91b3779703844825a5R187-R193

Reply:
@Vadman97 @ccschmitz - what do you think about removing that `useEffect` I listed above? I'm happy to do some rigorous click testing on this.

Reply:
oh i see! nice find eric - removing that `useEffect` definitely makes more sense. with that `useEffect`, `submitQuery('')` causes a state update, which triggers the `formState.useSubmit`. my method of checking `formState.submitting` determines whether this `submitQuery` comes from this manual call vs the form being updated by an input, but that additional indirection is certainly a worse approach. updating this